Docs: Document fine-grained access control (#33563)
grafana/grafana: docs/sources/enterprise/access-control/_index.md

+++
title = "Fine-grained Access control"
description = "Understand how to grant, change or revoke access to the Grafana resources"
keywords = ["grafana", "fine-grained-access-control", "roles", "permissions", "enterprise"]
weight = 100
+++

> # Fine-grained access control

Author: This is a bit tricky to do, as essentially everyone who has required permissions can manage access control. That said, when Grafana starts, in the beginning only Grafana Server Admin can do that, and Grafana Server Admin can later decide that someone else needs also to be able to do it and give them permissions.

I will look into where it would make sense to mention this somehow.

> The fine-grained access control is currently in beta. Expect changes in future releases.

> Fine-grained access control is available behind the `accesscontrol` feature flag in Grafana Enterprise 8.0+.

> Fine-grained access control provides a standardized way of granting, changing and revoking access to Grafana resources. 

Reviewer: Agree with @vtorosyan's sentiment. Perhaps where we use the term fine-grained access control other than the title and definition, we can find more specific terms like "custom roles" or "updating permissions for a role."

The fine-grained access control works hand-in-hand with the existing [Grafana permissions]({{< relref "../../permissions/_index.md" >}}) and allows you to control actions users can perform in a granular way.

> To learn more about how fine-grained access control works, refer to [Concepts]({{< relref "./concepts/_index.md" >}}) and to [Manage roles and permissions]({{< relref "./managing-roles-permissions.md" >}}).

Author: Honestly the easiest "fix" would be to remove this hierarchy and have flat structure, so roles, permissions and custom roles will be in the same level as the index 🤔

+++
title = "Fine-grained access control"

> description = "Understand how to grant, change or revoke access to the Grafana resources"

Reviewer 2: ```suggestion
description = "Grant, change, or revoke access to Grafana resources"
```

keywords = ["grafana", "fine-grained-access-control", "roles", "permissions", "enterprise"]
weight = 100
+++

# Fine-grained access control

> > **Note:** Fine-grained access control is currently in beta. Expect changes in future releases.

Reviewer 2: ```suggestion
> **Note:** Fine-grained access control is in beta, and you can expect changes in future releases.
```

Author: One day, would be nice to update these in other docs. Since I suspect many people, including me are just looking at the previous examples and copy pasting it.

> Fine-grained access control provides a standardized way of granting, changing, and revoking access to view and modify Grafana resources, like users and reports. 

Reviewer 2: ```suggestion
Fine-grained access control provides a standardized way of granting, changing, and revoking access when it comes to viewing and modifying Grafana resources, such as users and reports. 
```

> Fine-grained access control works hand-in-hand with the existing [Grafana permissions]({{< relref "../../permissions/_index.md" >}}) and allows you to control actions users can perform in a more granular way.

Reviewer 2: ```suggestion
Fine-grained access control works with [Grafana permissions]({{< relref "../../permissions/_index.md" >}}), and it allows you granular control of users’ actions.
```

To learn more about how fine-grained access control works, refer to [Roles]({{< relref "./roles.md" >}}) and [Permissions]({{< relref "./permissions.md" >}}).

> Refer to [fine-grained access control usage scenarios]({{< relref "./usage-scenarios.md" >}}) to see step-by-step instructions and understand how you can use the fine-grained access control system.

Reviewer 2: ```suggestion
To use the fine-grained access control system, refer to [Fine-grained access control usage scenarios]({{< relref "./usage-scenarios.md" >}}).
```

## Enable fine-grained access control

Fine-grained access control is available behind the `accesscontrol` [feature toggle]({{< relref "../../administration/configuration.md#feature_toggles" >}}) in Grafana Enterprise 8.0+.

> Refer to [Configuration]({{< relref "../../administration/configuration.md" >}}) to understand how you can enable it in different environments.

Reviewer 2: ```suggestion
To enable it in different environments, refer to [Configuration]({{< relref "../../administration/configuration.md" >}}).
```

## Access management

> When making a decision about access, the fine-grained access control takes into the account the following main parts:
> - _who_ has an access (`identity`).
> - _what they can do_ and on which _Grafana resource_ (`role`).

Reviewer 2: ```suggestion
Fine-grained access control considers a) _who_ has an access (`identity`), and b) _what they can do_ and on which _Grafana resource_ (`role`).
```

> You can grant, change, or revoke access to _users_ (`identity`). When an authenticated user tries to access a Grafana resource, the authorization system checks the required [fine-grained permissions]({{< relref "./permissions.md" >}}) for the resource and determines whether the action is allowed or not. 

Reviewer 2: ```suggestion
You can grant, change, or revoke access to _users_ (`identity`). When an authenticated user tries to access a Grafana resource, the authorization system checks the required fine-grained permissions for the resource and determines whether or not the action is allowed. Refer to [Fine-grained permissions]({{< relref "./permissions.md" >}}).
```

> To grant or revoke access to your users, you can create or remove a [built-in role assignments]({{< relref "./roles.md#built-in-role-assignments" >}}).

Reviewer 2: ```suggestion
To grant or revoke access to your users, create or remove built-in role assignments. For more information, refer to [Built-in role assignments]({{< relref "./roles.md#built-in-role-assignments" >}}).
```

> To grant or revoke access to your users, you can create or remove a [built-in role assignments]({{< relref "./roles.md#built-in-role-assignments" >}}).

Reviewer 2: ```suggestion
To grant or revoke access to your users, create or remove built-in role assignments. For more information, refer to [Built-in role assignments]({{< relref "./roles.md#built-in-role-assignments" >}}).
```

## Resources with fine-grained permissions

> Fine-grained access control is in beta and [permissions]({{< relref "./permissions.md" >}}) are available only for a subset of resources.

Reviewer 2: ```suggestion
Fine-grained access control permissions are available for a subset of resources. For more information, refer to [Permissions]({{< relref "./permissions.md" >}}).
```

> Refer to the relevant API guide from below list to learn more about specific endpoints where you can use access control.
> 
> - [Access Control API]({{< relref "../../http_api/access_control.md" >}})
> - [Admin API]({{< relref "../../http_api/admin.md" >}})
> - [Organization API]({{< relref "../../http_api/org.md" >}})
> - [Reporting API]({{< relref "../../http_api/reporting.md" >}})
> - [User API]({{< relref "../../http_api/user.md" >}})

Reviewer 2: ```suggestion
To learn more about specific endpoints where you can use access control, refer to the relevant API guide:

- [Access Control API]({{< relref "../../http_api/access_control.md" >}})
- [Admin API]({{< relref "../../http_api/admin.md" >}})
- [Organization API]({{< relref "../../http_api/org.md" >}})
- [Reporting API]({{< relref "../../http_api/reporting.md" >}})
- [User API]({{< relref "../../http_api/user.md" >}})
```
